Client/mods/deathmatch/logic/lua/CLuaFunctionDefs.World.cpp

```diff
@@ -1920,3 +1922,139 @@ int CLuaFunctionDefs::FetchRemote(lua_State* luaVM)
+
+    lua_newtable(luaVM);
+    for (const auto& request : g_pClientGame->GetRemoteCalls()->GetCalls())
+    {
+        if(!pLuaMain || request->GetVM() == pLuaMain)
+        {
+            lua_pushnumber(luaVM, ++iIndex);
+            lua_pushuserdata(luaVM, request);
+            lua_settable(luaVM, -3);
+        }
+    }
+
+    return 1;
+}
+
+// table getRemoteRequestInfo(element theRequest)
+int CLuaFunctionDefs::GetRemoteRequestInfo(lua_State* luaVM)
+{
+    CScriptArgReader argStream(luaVM);
+    CLuaArguments    info, requestedHeaders;
+    CRemoteCall*     pRemoteCall = nullptr;
+    CResource*       pThisResource = g_pClientGame->GetResourceManager()->GetResourceFromLuaState(luaVM);
+
+
+    argStream.ReadUserData(pRemoteCall);
+
+    if (!argStream.HasErrors())
+    {
+        CResource* pResource = pRemoteCall->GetVM()->GetResource();
+        bool bExtendedInfo = (pResource == pThisResource);
+        
+        info.PushString("type");
+        info.PushString((pRemoteCall->IsFetch() ? "fetch" : "call"));
+
+        // TODO: remove query_string from url when bExtendedInfo isn't set
+        info.PushString("url");
+        info.PushString(pRemoteCall->GetURL().c_str());
+
+        info.PushString("queue");
+        info.PushString(pRemoteCall->GetQueueName().c_str());
+        
+        info.PushString("resource");
+        
+        if (pResource)
+            info.PushResource(pResource);
+        else
+            info.PushBoolean(false);
+
+        info.PushString("start");
+        info.PushNumber(pRemoteCall->GetStartTime());
+
+        if (bExtendedInfo)
+        {
+            info.PushString("postData");
+            info.PushString(pRemoteCall->GetOptions().strPostData.c_str());
+
+            // requested headers
+            info.PushString("headers");
+
+            for (auto const& header : pRemoteCall->GetOptions().requestHeaders)
+            {
+                requestedHeaders.PushString(header.first.c_str());
+                requestedHeaders.PushString(header.second.c_str());
+            }
+
+            info.PushTable(&requestedHeaders);
+        }
+
+        info.PushString("method");
+        info.PushString((pRemoteCall->GetOptions().strRequestMethod.length() >= 1 ? pRemoteCall->GetOptions().strRequestMethod.ToUpper().c_str() : "POST"));
+
+        info.PushString("connectionAttempts");
+        info.PushNumber(pRemoteCall->GetOptions().uiConnectionAttempts);
+
+        info.PushString("connectionTimeout");
+        info.PushNumber(pRemoteCall->GetOptions().uiConnectTimeoutMs);
+
+        // download info
+        SDownloadStatus downloadInfo;
+
+        if (pRemoteCall->GetDownloadStatus(downloadInfo))
+        {
+            info.PushString("bytesReceived");
+            info.PushNumber(downloadInfo.uiBytesReceived);
+
+            info.PushString("bytesTotal");
+            info.PushNumber(downloadInfo.uiContentLength);
+        }
+
+        info.PushString("currentAttempt");
```

Comment:
Should this be inside previous `if{ ... }` block?

Reply:
Are you talking about `currentAttempt` ? If so, it's always set which is why it's not inside the if block condition.

Reply:
Yes, I was talking about `currentAttempt`.
However, after looking at `CRemoteCall::GetDownloadStatus()` I realized that returning false can be avoided, so it has been refactored to return `const SDownloadStatus&` instead. You'll have to update this section of code to cope with that change.